Sources/OpenAI/Public/Models/ChatResult.swift

```diff
@@ -10,13 +10,20 @@ import Foundation
 public struct ChatResult: Codable, Equatable {
     
     public struct Choice: Codable, Equatable {
+        public struct Delta: Codable, Equatable {
+            public let content: String?
+            public let role: Chat.Role?
+        }
+        
         public let index: Int
-        public let message: Chat
-        public let finishReason: String
+        public let message: Chat?
```

Comment:
If you want to mix the complete and partial results, can you comment in what condition these properties are `nil`?

I have moderate preference in separating a complete `Choice` from a `PartialChoice`, provided that they have major differences, despite the official OpenAI API mix them together.

```swift
// Choice stays the same

public struct PartialChoice: Codable, Equatable {
  public let index: Int
  public let delta: Delta
}

```

Reply:
I've decided to keep `ChatResult`to not break backward compatibility and created "ChatStreamingResult" along with a separate function for streaming. I found this solution more straightforward and clean. What do you think about this?